### PR TITLE
Report linter errors as the main PR review comment

### PR DESCRIPTION
**Pull request by gylaz**
When a linter encounters an unexpected error, like invalid config
syntax, we will receive it in the `error` attribute in the payload.
Then we'll report it to the user via the main comment that goes into the
PR review body.

### CODE REVIEW

**Comment by salbertson** [comment]
@gylaz do you have a screenshot, or a test PR, to get a better idea what this would look like?

File: app/models/pull_request.rb [hunk 1]
```diff
@@ -62,6 +67,21 @@ def build_comment(violation)
...
+  def format_errors(errors)
+    if errors.any?
+      error_details = errors.map { |error| build_error_details(error) }
+      ["Some files could not be reviewed due to errors:"].
+        concat(error_details).join
+    end
+  end
+
+  def build_error_details(error)
+    lines = error.split("\n")
```
Comment (thorncp) [inline app/models/pull_request.rb]: `String#lines`?

File: spec/models/build_spec.rb [hunk 2]
```diff
@@ -37,4 +37,21 @@
       end
     end
   end
+
+  describe "#review_errors" do
+    it "returns and list of unique linter errors" do
```
Comment (thorncp) [inline spec/models/build_spec.rb]: and -> a?

**Comment by gylaz** [comment]
@salbertson @thorncp from this PR: https://github.com/gylaz/wormbots/pull/13#pullrequestreview-37914344
screenshot 2017-05-12 11 42 03

It will surface nearly identical errors, because it will be an error per file (and their body differs in the temp directory path). We could probably add some logic to do a 90% match on the error body.

**Pull request merged**